main_routers/mmd_router.py

```
@@ -91,80 +93,116 @@ def _ensure_mmd_directory(config_mgr) -> Path | None:
+            raw_bytes_blob.decode(fallback_enc)
+            return fallback_enc
+        except UnicodeDecodeError:
             continue
 
     return None
 
 
-def _build_zip_name_map(zf: zipfile.ZipFile) -> dict[str, str]:
-    """为 ZIP 中所有条目构建「原始名 → 正确解码名」的映射表。
+def _sanitize_filename(filename: str) -> str:
+    """清理文件名，仅移除操作系统不允许的非法字符，保留中日文和路径结构。"""
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)